lib/ansible/executor/task_executor.py

```diff
@@ -824,6 +817,33 @@ def _evaluate_failed_when_result(result):
         display.debug("attempt loop complete, returning result")
         return result
 
+    @staticmethod
+    def _retries_requested(_task):  # type: (Task) -> bool
+        return (_task.retries is not None) or bool(_task.until)
+
+    @staticmethod
+    def process_retry_parameters(retries_requested, _task):  # type: (bool, Task) -> Tuple[int, int]
+        '''
+        Extract the parameters used in retrying the task
+        '''
+        attempts = 1
+        if retries_requested:
+            if _task.retries is not None:
+                attempts += _task.retries
+            else:
+                attempts += 3
+
+        if attempts <= 0:  # guard against negative retries
+            attempts = 1
+
+        delay = _task.delay
+        if delay is None:
+            delay = _task._delay.default
```

Comment:
this wont work for 2 reasons a) you removed the default from the definition b) field attributes have changed so this is accessing a non existant method

Reply:
I'm not sure I understand what you're saying. The default for the `default` parameter is `None`. The attribute I removed the default from is `_retries`, not `_delay`. There is already a test that a task with an `until` section but not `retries` has the default of 3, and it is still passing.
But yeah, I guess a lot changes in a year. I'll see if this rebases.

Reply:
oh, the new implementation of `FieldAttribute` doesn't support introspection. The builtin `Property` has this introspectability:
```python
>>> class C:
...     @property
...     def p(self):
...         ...
... 
>>> C.p
<property object at 0x7fc10158b6f0>
```
adding this is simple, in `FieldAttribute.__get__` we just need the line [ref](https://docs.python.org/3/howto/descriptor.html#properties)
```python
    if obj is None:
        return self
```
This enables programatically accessing the default of the `FieldAttribute`, making tests less brittle. Alternatively, `FieldAttribute` coincidentally returns the default if invoked when unbound, so I can just access that in a test and hope it doesn't break in the future. Or I can just hardcode the expected value of the default and things will break if that ever changes.